CONTRIBUTING.md

```diff
@@ -97,137 +76,312 @@ script before each commit and abort the commit on error. To
+screen, a prompt will be displayed: “Folder contains a Dev Container
+configuration file. Reopen folder to develop in a container.”.
+- Selecting the “Reopen in Container” option will automatically start the
+process of creating the Dev Container; this may take a few minutes the first
+time the container is created, or any time the settings for the container have
+changed and the container needs to be recreated.
+- If there are issues running the `ci.sh` script in a clean checkout, check the
+`Files: Eol` setting in Visual Studio Code. Change it to \n, open the `ci.sh`
+file, save it, and retry the script.
 
-Test changes by directly loading changes locally, without having to re-package and re-publish the extension in the marketplace.
+### Windows Subsystem For Linux
 
-Reference: [Azure DevOps Extension Hot Reload and Debug](https://github.com/microsoft/azure-devops-extension-hot-reload-and-debug)
+1. To configure the WSL if it is not already available on the development
+machine, follow this [tutorial](https://docs.microsoft.com/en-us/learn/modules/get-started-with-windows-subsystem-for-linux/).
 
-**Note:** You will need [Visual Studio Code](https://code.visualstudio.com/download), [Firefox](https://www.mozilla.org/en-US/firefox/) and the [Debugger for Firefox](https://marketplace.visualstudio.com/items?itemName=firefox-devtools.vscode-firefox-debug) VS Code extension.
+2. Follow [Git on WSL Instructions](https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-git) if Git is not already installed on the development machine.
 
-- In the 'RetrospectiveExtension.Frontend' folder, create the 'vss-extension-dev.json' file using the template file `vss-extension-dev.json.template` for reference.
+3. Follow [Node Installation Instructions](https://docs.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-wsl) to enable Node for WSL.
 
-- Update the 'webpack.config.js' to enable source maps. Set the devtool property to `inline-source-map`. Also set devServer.https to true and devServer.port to 3000.
+4. Install the latest version of [Visual Studio Code](https://code.visualstudio.com/).
 
-  ```js
-  module.exports = {
-    devtool: 'inline-source-map',
-    devServer: {
-      https: true,
-      port: 3000,
-      static: {
-        directory: path.join(__dirname),
-      }
-    },
-  ...
-  ```
+5. Install the [Remote – WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) extension for Visual Studio Code.
 
-- Set `output.publicPath` to `/dist/` in the webpack.config.json file. This will allow webpack to serve files from `https://localhost:3000/dist`.
+6. Check out this repository and open the parent folder in Visual Studio Code.
 
-  ```js
-  module.exports = {
-      output: {
-        publicPath: "/dist/"
-        // ...
-      }
-      // ..
-  };
-  ```
+7. The first time the repository is opened in Visual Studio Code, the terminal
```

Comment:
Maybe phrasing it as a `If VSCode does not recognize the WSL on its own, ....` so the step feels like it's "skippable" if it doesn't apply to the developer.